zone/attack.cpp

```diff
@@ -1522,6 +1522,27 @@ bool Client::Attack(Mob* other, int Hand, bool bRiposte, bool IsStrikethrough, b
 
 	other->AddToHateList(this, hate);
 
+	//Guard Assist Code
+	if (RuleB(Character, PVPEnableGuardFactionAssist) && this->IsClient() || RuleB(Character, PVPEnableGuardFactionAssist) && this->HasOwner() && this->GetOwner()->IsClient()) {
+		auto& mob_list = entity_list.GetCloseMobList(other);
+		for (auto& e : mob_list) {
+			auto mob = e.second;
+			float distance = Distance(other->CastToClient()->m_Position, mob->GetPosition());
+			if (mob->CheckLosFN(other) && distance <= 70 || mob->CheckLosFN(this) && distance <= 70) {
+				if (mob->CastToNPC()->GetNPCFactionID() == 382 || mob->CastToNPC()->GetNPCFactionID() == 778 || IsGuard(mob->GetRace(),mob->GetTexture())) {
```

Comment:
What are these faction ID's supposed to be ? These should at least tie back to a constant of some kind and pulling these values out of the database to remove magic numbers from the logic

Something like

```cpp
// database factions
const int16 DB_FACTION_WATCHMEN       = 382;
const int16 DB_FACTION_PAINEEL_GUARDS = 778;
```

And at least put it in somewhere like `factions.h` for now

Reply:
Also if these factions are guards then it could fall underneath your `IsGuard` method and factions can just be an additional check there. 

You don't need to pass parameters because it's already a mob method and you have access to mob-scoped methods and there's not a huge benefit to explicitly passing the parameters